src/Repository/Storable.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Friendica\Repository;
+
+use Friendica\Collection;
+use Friendica\Model;
+use Friendica\Repository;
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Network\HTTPException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Storable repositories are linked to a main table with an id primary key and will instanciate Models and Collections
+ * related to that table.
+ */
+abstract class Storable extends Repository
```

Comment:
I'm missing a `save` possibility, where one or more models are possible to get updated/inserted.

Reply:
Yes, I've not gotten to that point yet. I think I'll add a field array parameters to manually limit the number of fields saved. I'm not sure I like your changed field optimization yet.

Reply:
Why you don't like it? I thought maybe in case we just changed on field inside `item` instance, we maybe have to unnecessary update all other fields as well .. Or in case we "accidentally" didn't changed a field, but call "update" without a need.

Or is such functionality already implemented inside the DB class itself? (Like checking if something changed before updating it?) Because then => nevermind :D

Reply:
Not in the DB class itself that I know of. I don't like it because it's implicit, on top of the performance hit of the magic method `__get` and probably `__set` too.

Reply:
A known alternative would be using `getter` and `setter` for the attributes and adding the change logic to each `setter`. Hmmm.. The performance impact is indeed a problem.. I didn't know it until now..